packaging/genere-views.cmake

```diff
@@ -0,0 +1,78 @@
+# Launch genere binary
+# cmake-format: off
+# The funtion api is almost the same as genere:
+#
+# Genere <ClassName> <ClassLabel> <AttributeFileName> <LogFile> [options]
+#   <ClassName>: base name for generated classes
+#   <ClassLabel>: label for generated classes
+#   <AttributeFileName>: name of the file (.dd) containing the attribute specifications
+#   <LogFile>: name of the log file, it must be included in the source files of the binary/library
+#
+# Options:
+#   -nomodel no generation of class <ClassName>
+#   -noarrayview no generation of class <ClassName>ArrayView
+#   -noview no generation of classes <ClassName>View and <ClassName>ArrayView
+#   -nousersection no generation of user sections
+#   -specificmodel <SpecificModelClassName> name of a specific model class, to use instead of ClassName
+#   -super <SuperClassName> name of the parent class
+#
+# Do nothing if GENERATE_VIEWS is set to OFF
+#
+# Example to use genere binary in CMakeLists.txt:
+#
+#   include(genere)
+#   generate_gui(FOO "Foo Example" Foo.dd Foo.dd.log -noarrayview)
+#   add_executable(basetest Foo.cpp) 
+#   if(GENERATE_VIEWS)
+#       target_sources(basetest PRIVATE Foo.dd.log) # Foo.dd.log is genereted by generate_gui
+#   endif()
+#   generate_gui_add_dependency(basetest)
+#
+# The log file Foo.dd.log is generated by generate_gui and must be added as a source of the target. 
+# Whithout it, generate_gui is never triggered.
+#
+# cmake-format: on
+function(generate_gui ClassName ClassLabel AttributeFileName LogFile)
+
+  if(GENERATE_VIEWS)
+    find_package(Python REQUIRED)
+    add_custom_command(
+      OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/${LogFile}
+      COMMAND
+        ${Python_EXECUTABLE} ${CMAKE_SOURCE_DIR}/packaging/genere-script.py ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/genere
+        ${ARGN} ${ClassName} ${ClassLabel} ${AttributeFileName} ${LogFile}
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+      VERBATIM
+      DEPENDS ${AttributeFileName}
+      MAIN_DEPENDENCY ${AttributeFileName}
+      COMMENT "Generating ${AttributeFileName} views")
+  endif()
+endfunction()
+
+# Create dependency between the target parameter and the genere binary. This function should be used with generate_gui
```

Comment:
You can avoid having this function if in `add_view_generation` you add a `${ParentTargetName}` parameter. Then at the end of that function you execute
```cmake
add_dependencies(${ParentTargetName} genere)
```
This way you wont risk of forgetting calling the dependecy function.

Note that there is no problem if you execute many times the same `add_dependencies` statement.

Reply:
I would add also a status message for when we create the configuration
```cmake
 # Show status  message
 message(STATUS "Setting up VIEW_GENERATION for class ${ClassName} in ${ParentTargetName}")
```

Reply:
There is not function `add_view_generation` I think you talk about `generate_gui`. We have discuss this before with Marc and we think it's better to keep it as it is: each function is doing only one thing.
For the message, I'm not sure it would bring any information and furthermore it would generate a line for each call of `generate_gui` . It is call 25 times only in the KWUserInterface project

Reply:
Sorry, I forgot to add the name change the name suggestion. In general functions in CMake add actions to the build process: generate_gui doesn't generate the gui, it adds the gui generation to the build, so the name is better.

The message assures us that the target was added. It is lengthy but we make it would only be present when `GENERATE_VIEWS=ON`.

As for the dependencies: if you add a gui generation you **must** add the dependency. Why risking forget it ?Furthermore it simplifies the code. I do not see the downside.

Reply:
OK I want to keep the same prefix for all methods of the module, it would become  `generate_gui_add_view`.
Again I disagree for the message, if we add 100 messages there is no information at all...
For the dependency, I let you see with @marcboulle

Reply:
Ok for the name changing, and no messages.

After talk with @marcboulle we'll keep the current function structure.